addons/rating/models/mail_thread.py

```diff
@@ -1,12 +1,190 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models, _
+import datetime
+import markupsafe
+
+from odoo import api, fields, models, _, tools
 
 
 class MailThread(models.AbstractModel):
     _inherit = 'mail.thread'
 
+    rating_ids = fields.One2many('rating.rating', 'res_id', string='Rating', groups='base.group_user',
+                                 domain=lambda self: [('res_model', '=', self._name)], auto_join=True)
+
+    # MAIL OVERRIDES
+    # --------------------------------------------------
+
+    def unlink(self):
+        """ When removing a record, its rating should be deleted too. """
+        record_ids = self.ids
+        result = super(MailThread, self).unlink()
+        self.env['rating.rating'].sudo().search([('res_model', '=', self._name), ('res_id', 'in', record_ids)]).unlink()
+        return result
+
+    def _message_create(self, values_list):
+        """ Force usage of rating-specific methods and API allowing to delegate
+        computation to records. Keep methods optimized and skip rating_ids
+        support to simplify MailThrad main API. """
+        if not isinstance(values_list, (list)):
+            values_list = [values_list]
+        if any(values.get('rating_ids') for values in values_list):
+            raise ValueError(_("Posting a rating should be done using message post API."))
+        return super()._message_create(values_list)
+
+    # RATING CONFIGURATION
+    # --------------------------------------------------
+
+    def _rating_apply_get_default_subtype_id(self):
+        return self.env['ir.model.data']._xmlid_to_res_id("mail.mt_comment")
+
+    def _rating_get_operator(self):
+        """ Return the operator (partner) that is the person who is rated.
+
+        :return record: res.partner singleton
+        """
+        if hasattr(self, 'user_id') and self.user_id.partner_id:
```

Comment:
it should do the same
```suggestion
        if 'user_id' in self and self.user_id.partner_id:
```

Reply:
yes tested with the shell and changed, thanks